packages/api/src/UserPreferences/UserPreferences.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { UserPreferences, UserPreferencesInput } from './UserPreferences.entity';
+
+@Injectable()
+export class UserPreferencesService {
+  constructor(
+    @InjectRepository(UserPreferences)
+    private readonly userPreferencesRepository: Repository<UserPreferences>
+  ) { }
+
+  async findByUser(userId: string): Promise<UserPreferences | undefined> {
+    const userPreference = await this.userPreferencesRepository.findOne({ where: { userId } });
+    return userPreference;
+  }
+
+  async update(userId: string, input: UserPreferencesInput): Promise<UserPreferences> {
+    const userPreferences = await this.userPreferencesRepository.findOne({ where: { userId } });
+    if (userPreferences) {
+      return this.userPreferencesRepository.save({ ...userPreferences, ...input });
+    }
+    return this.userPreferencesRepository.create({ userId, ...input }).save();
+  }
```

Comment:
Is there an upsert in typeorm? If so let's us this, otherwise ignore this comment

Reply:
I didn't see any